File: trunk/quote/Projects/PluginModel/template/DefaultComponentProperties.vb
Imports System.ComponentModel
Imports System.Reflection

Namespace Template

    Public Class DefaultComponentProperties
        Inherits Template.ComponentProperties

        Private _QuoteDetail As Template.Detail

        Public Sub New(ByVal quoteDetail As Template.Detail)
            MyBase.New(quoteDetail)
            _QuoteDetail = quoteDetail
        End Sub

        Protected Overloads Sub SendEvents()
            MyBase.SendEvents()
            Me._QuoteDetail.Header.ComputationProperties.SendEvents()
        End Sub

    End Class
End Namespace

File: trunk/quote/Projects/PluginModel/template/DefaultComponentProperties.cs
namespace Model.Template
{
    using System;
    using System.Linq;

    /// <summary>
    /// The default component properties.
    /// </summary>
    public sealed class DefaultComponentProperties : Template.ComponentProperties
    {
        /// <summary>
        /// Initializes a new instance of the <see cref="DefaultComponentProperties" /> class.
        /// </summary>
        /// <param name="quoteDetail">The detail object.</param>
        public DefaultComponentProperties(Template.Detail quoteDetail) : base(quoteDetail)
        {
        }
    }
}
